### PR TITLE
Expose Ip-Family for IPV6 (#1089)

### PR DESCRIPTION
**Pull request by rquitales**
### Proposed changes
This PR exposes the `ipFamily` argument to enable creating EKS clusters that support ipv6 networking.
Modified from https://github.com/pulumi/pulumi-eks/pull/888 and rebased on top of latest changes in master.

An additional cluster with ipv6 is also created to test that cluster creation works as expected.

### Related issues (optional)
Closes: #698
Closes: #684

### CODE REVIEW

**Comment by davidroth** [comment]
@rquitales It looks like the acceptance tests have been successfully completed.🥳
Happily awaiting this finally being merged (I absolutely need this change as soon as possible for a new project.) 🚀

File: examples/cluster-go/main.go [hunk 1]
```diff
@@ -39,10 +45,79 @@ func main() {
...
+
+			subnetIDs = append(subnetIDs, subnet.ID().ToStringOutput())
+		}
+
+		// 3. Create an EKS cluster with IPv6 networking enabled.
+		cluster4, err := eks.NewCluster(ctx, "example-cluster-4", &eks.ClusterArgs{
+			IpFamily:         pulumi.StringPtr("ipv6"),
+			VpcId:            vpc.ID().ToStringOutput(),
+			SubnetIds:        pulumi.StringArray(subnetIDs),
+			UseDefaultVpcCni: func() *bool { t := true; return &t }(),
```
Comment (EronWright) [inline examples/cluster-go/main.go:89]: This line seems unusual, could it be `pulumi.BoolPtr(true)`?

File: examples/cluster-go/main.go [hunk 2]
```diff
@@ -28,6 +32,8 @@ func main() {
 			return err
 		}
 
+		// Create cluster that don't conflict between defaults and mutually exclusive arguments
```
Comment (EronWright) [inline examples/cluster-go/main.go:35]: Can you rephrase this comment?

**Review by EronWright — APPROVED** [review]

File: examples/cluster-go/main.go [hunk 3]
```diff
@@ -39,10 +45,79 @@ func main() {
...
+		var subnetIDs []pulumi.StringInput
+		for idx, az := range []string{"us-west-2a", "us-west-2b"} {
+			subnet, err := ec2.NewSubnet(ctx, fmt.Sprintf("ipv6-subnet-%d", idx), &ec2.SubnetArgs{
+				VpcId:                       vpc.ID().ToStringOutput(),
+				AssignIpv6AddressOnCreation: pulumi.Bool(true),
+				// EnableDns64:                             pulumi.Bool(true),
+				// EnableResourceNameDnsAaaaRecordOnLaunch: pulumi.Bool(true),
+				AvailabilityZone: pulumi.String(az),
+				CidrBlock:        pulumi.String(fmt.Sprintf("10.100.%d.0/24", len(subnetIDs))),
+				// Ipv6Native:    pulumi.Bool(true), // Create a ipv6 only subnet
```
[the remark this answers is not in the document]
Comment (rquitales) [reply examples/cluster-go/main.go:74]: This was me trying out different configuration options. I'll remove it since it's not necessary.

File: examples/cluster-go/main.go (hunk 1 above)
Comment (rquitales) [reply examples/cluster-go/main.go:89]: `UseDefaultVpcCni` requires a `*bool` type, and that's because we define it as a plain type in schema (ref: https://github.com/pulumi/pulumi-eks/blob/449635cb9dfdfbdc6a03b2d0c3b8b12f96420f13/provider/cmd/pulumi-gen-eks/main.go#L319) when the MLCs were created for this provider. This is an artifact of the nodejs types driving the MLC types, and ideally should really not be a plain type here.

**Comment by rquitales** [comment]
Note: the previously passed all pre-submits. A new commit was added that only modified comments (https://github.com/pulumi/pulumi-eks/pull/1098/commits/6c3675e153ee6a729ef26975415bff49988a2e88), so I'll be force pushing it since retesting requires another 30-45 mins to run. We also have post-submit CI runs that also exercise the same tests.
